syntheseus/search/algorithms/base.py

```diff
@@ -386,7 +387,8 @@ def _filter_reactions(
             reactions = [
                 rxn
                 for rxn in reactions
-                if frozenset((node.mols - {rxn.product}) | rxn.reactants) not in ancestor_molsets
+                if frozenset((node.mols - {rxn.product}) | set(rxn.reactants))
```

Comment:
Your suggested change (and your similar suggestions below) would work and indeed avoid a redundant call to `set`. However, _it fail the mypy check_ because I defined `unique_reactants` to return `Collection[Molecule]` instead of `set[Molecule]`. Collection objects do not necessarily support the `|` operator (see table [here](https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes)). Therefore mypy raises an `operator` error.

We could either:
1. Leave it as it is and accept the small redundancy
2. Put a `type: ignore[operator]` comment to make mypy happy
3. Narrow the type of `unique_reactions` from `Collection[Molecule]` to `set[Molecule]`

What do you think is the best option?

Reply:
I think (3) may be the best option; I don't see a good reason to restrict the type hint to `Collection`.